src/Manifolds.jl

```diff
@@ -435,68 +435,119 @@ function shortest_geodesic(M::Manifold, x, y, T::AbstractVector)
+
+"""
+    ProjectTangent <: AbstractVectorTransportMethod
+
+Specify to use projection onto tangent space as vector transport method within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). See [`project_tangent`](@ref) for details.
+"""
+struct ProjectTangent <: AbstractVectorTransportMethod end
+
+"""
+    vector_transport_to!(M::Manifold, vto, x, v, y [, m])
 
 Vector transport of vector `v` at point `x` to point `y`. The result is saved
-to `vto`. By default, [`project_tangent!`](@ref) is used but this may change in
-the future.
+to `vto`. By default, the method `m` is [`ParallelTransport`](@ref).
 """
-vector_transport_to!(M::Manifold, vto, x, v, y) = project_tangent!(M, vto, y, v)
+vector_transport_to!(M::Manifold, vto, x, v, y) = vector_transport_to!(M,vto,x,v,y,ParallelTransport())
+vector_transport_to!(M::Manifold, vto, x, v, y, m::ProjectTangent) = project_tangent!(M, vto, y, v)
+
+function vector_transport_to!(M::Manifold, vto, x, v, y, m::AbstractVectorTransportMethod)
+    error("vector transport from a point of type $(typeof(x)) to a type $(typeof(y)) on a $(typeof(M)) for a vector of type $(v) and the $(typeof(m)) not yet implemented.")
+end
+
 
 """
-    vector_transport_to(M::Manifold, x, v, y)
+    vector_transport_to(M::Manifold, x, v, y[,method=::ParallelTransport])
 
-Vector transport of vector `v` at point `x` to point `y`.
+Vector transport of vector `v` at point `x` to point `y` using the method `m`,
+which defaults to [`ParallelTransport`](@ref).
 """
-function vector_transport_to(M::Manifold, x, v, y)
+vector_transport_to(M::Manifold, x, v, y) = vector_transport_to(M,x,v,y,ParallelTransport())
+function vector_transport_to(M::Manifold, x, v, y, m)
     vto = similar_result(M, vector_transport_to, v, x, y)
-    vector_transport_to!(M, vto, x, v, y)
+    vector_transport_to!(M, vto, x, v, y, m)
     return vto
 end
 
 """
-    vector_transport_direction!(M::Manifold, vto, x, v, vdir)
+    vector_transport_direction!(M::Manifold, vto, x, v, vdir[, m=::ParallelTransport])
 
 Vector transport of vector `v` at point `x` in the direction indicated
 by the tangent vector `vdir` at point `x`. The result is saved to `vto`.
-By default, `exp` and `vector_transport_to!` are used.
+By default, `exp` and `vector_transport_to!` are used with the method `m` which
+defaults to [`ParallelTransport`](@ref)..
 """
-function vector_transport_direction!(M::Manifold, vto, x, v, vdir)
+vector_transport_direction!(M::Manifold, vto, x, v, vdir) = vector_transport_direction!(M,vto,x,v,vdir,ParallelTransport())
+function vector_transport_direction!(M::Manifold, vto, x, v, vdir,m)
     y = exp(M, x, vdir)
-    return vector_transport_to!(M, vto, x, v, y)
+    return vector_transport_to!(M, vto, x, v, y, m)
 end
 
 """
-    vector_transport_direction(M::Manifold, x, v, vdir)
+    vector_transport_direction(M::Manifold, x, v, vdir[, m=::ParallelTransport])
```

Comment:
Should be `method=ParallelTransport()` or `method::AbstractVectorTransportMethod=ParallelTransport()`.